### PR TITLE
Fix/589/import issues

### LINKED ISSUE #589: Import from LastPass or KeePass impossible
### Steps to reproduce
1. Generate Keepas CSV file
2. Go to Import section in Passman parameters
3. Select Keepass import type
4. Start import

### Expected behaviour
All my password will be added in my Passman account

### Actual behaviour
No import and error in javascript console

### Configuration

**Browser**: Chromium 74.0.3729.131       

**Extensions that might cause interference**:   

**Passman version**:    2.3.0

**Operating system**: Windows 10

**Web server:** Apache (Nextcloud-Docker)

**Database:** MariaDB

**PHP version:**
      
**cloud server:** Nextcloud

**cloud version:** 16.0.0
  
   
#### Browser log
<details>
passman.min.js?v=806176c8-0:10 Uncaught TypeError: Cannot read property 'slice' of undefined
    at Object.encrypt (passman.min.js?v=806176c8-0:10)
    at Object.encrypt (passman.min.js?v=806176c8-0:11)
    at Object.encryptString (passman.min.js?v=806176c8-0:28)
    at Object.createCredential (passman.min.js?v=806176c8-0:28)
    at addCredential (passman.min.js?v=806176c8-0:31)
    at C_Promise.finally (passman.min.js?v=806176c8-0:31)
    at C_Promise.call_then (passman.min.js?v=806176c8-0:2)
    at C_Promise.<anonymous> (passman.min.js?v=806176c8-0:29)
encrypt @ passman.min.js?v=806176c8-0:10
encrypt @ passman.min.js?v=806176c8-0:11
encryptString @ passman.min.js?v=806176c8-0:28
createCredential @ passman.min.js?v=806176c8-0:28
addCredential @ passman.min.js?v=806176c8-0:31
(anonymous) @ passman.min.js?v=806176c8-0:31
C_Promise.call_then @ passman.min.js?v=806176c8-0:2
(anonymous) @ passman.min.js?v=806176c8-0:29
setTimeout (async)
C_Promise @ passman.min.js?v=806176c8-0:3
PassmanImporter.keepassCsv.readFile @ passman.min.js?v=806176c8-0:29
$scope.startImport @ passman.min.js?v=806176c8-0:31
(anonymous) @ passman.min.js?v=806176c8-0:6
b @ passman.min.js?v=806176c8-0:4
e @ passman.min.js?v=806176c8-0:7
$eval @ passman.min.js?v=806176c8-0:5
$apply @ passman.min.js?v=806176c8-0:5
(anonymous) @ passman.min.js?v=806176c8-0:7
dispatch @ main.js?v=806176c8-0:39
_.handle @ main.js?v=806176c8-0:39
</details>   

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/74174138-import-from-lastpass-or-keepass-impossible?utm_campaign=plugin&utm_content=tracker%2F44880056&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F44880056&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Oliv4945**
The `compromised` field of `credentials` object was not created by the importer, so there was an exception when encrypting the fields as `undefined` was pasted to the encryption library.

Should fix #589 

closes #589

### CODE REVIEW

File: js/importers/importer-keepasscsv.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ var PassmanImporter = PassmanImporter || {};
 		info: {
 			name: 'KeePass csv',
 			id: 'keepassCsv',
-			exportSteps: ['Create an csv export with the following options enabled: http://i.imgur.com/CaeTA4d.png']
+			exportSteps: ['If using Keepass V1: Create an csv export with the following options enabled: http://i.imgur.com/CaeTA4d.png', "With Keepass V2 or Keepass XC no configuration is needed']
```
Comment (newhinton) [inline js/importers/importer-keepasscsv.js]: there is a small error there, you started a string with " and ended it with ' ;) can you swap the " ? :D

**Comment by robin-thoni** [comment]
@Oliv4945 Thanks for the fix!

@newhinton That's a critical issue for people trying to migrate to passman. When can we expect this fix to be available? Thanks!

**Comment by newhinton** [comment]
it is merged, and will be shipped with the next release. However, there is no specific date when it will be released. But i guess it wont take long, maybe next week or so

**Comment by robin-thoni** [comment]
Sounds fair. Thanks again!

**Comment by newhinton** [comment]
You're welcome ;)

**Comment by robin-thoni** [comment]
Any news on updating the chrome app...?

"Updated
September 10, 2018"

That sounds a little bit old, although I totally understand it might not be your full time job. However, I think a few of us could be interested in collaborating and increasing the release cycle :) Currently the project sounds dead, which is a shame, as it is for me a really promising password manager.

**Comment by newhinton** [comment]
This project is not dead, we releases 2.3.2 just monday

are there issues with the current passman extension that require an update? while there are features that are not released for the extension, it is quiet stable afaik

**Comment by robin-thoni** [comment]
> This project is not dead
Ah! That's cool to hear that :)

> are there issues with the current passman extension that require an update?
Not that I'm aware of.

> while there are features that are not released for the extension, it is quiet stable afaik
I just realized this thread was about backend... Yeah, I meant updating the Nextcloud app, sorry... Friday evening... That's not the subject here, but as you're asking, the workflow to add a vault is "terrible" for two points:
- Leaving the focus will cancel adding a vault. That's more a Chrome (Firefox too? Didn't tested) problem, but that could be mitigated by allowing to use the same (without actually erasing all settings and resetting master password) page as the first setup. Use case is I use random passwords for vaults and an app password for login (because of U2F). That means I currently have to write at least of the passwords by hand, cause it is not possible to leave the focus to copy the second one.
- We have to login x times to add x vaults from the same account. Having a concept of accounts and vaults related to those would seriously simplify that workflow. Also, I store my other vaults passwords in my main one. Having a built-in feature to automatically use them would be even more awesome (for example, convention could be "passman-vault-${vault-name-or-id}".

Passwords are critical, and I have troubles to implement a user friendly workflow for my users, which is why I'd like to see those features. As it is critical to me, I'm ready to implement (some of) those features, and helping as much as I can, if we can have regular releases (when there's new features, of course).

**Comment by newhinton** [comment]
if you want to, you can always check out github.com/nextcloud/passman-webextension

(also, please file issues/features there, they will be overlooked here)

**Pull request merged**